src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala

```diff
@@ -867,20 +858,25 @@ class EthService(
       }
   }
 
-  private def withAccount[T](address: Address, blockParam: BlockParam)(f: Account => T): Either[JsonRpcError, T] = {
-    resolveBlock(blockParam).map { case ResolvedBlock(block, _) =>
-      f(
-        blockchain.getAccount(address, block.header.number).getOrElse(Account.empty(blockchainConfig.accountStartNonce))
-      )
+  private def withAccount[T](address: Address, blockParam: BlockParam)(f: Account => T): ServiceResponse[T] =
+    Future {
+      resolveBlock(blockParam).map { case ResolvedBlock(block, _) =>
+        f(
```

Comment:
Minor: I would like to name this function `toResponse` or sth similar and reformat the code to
```
val account = blockchain
            .getAccount(address, block.header.number)
            .getOrElse(Account.empty(blockchainConfig.accountStartNonce))
toResponse(account)
```
it is a little bit more readable